core/web/api/attached_files.py

```diff
@@ -6,12 +8,17 @@
 from core.web.api.api import render
 
 
-class Files(FlaskView):
+class AttachedFiles(FlaskView):
     def post(self):
-        if 'file' not in request.files:
-            abort(400)
+        if 'file' in request.files:
+            f = AttachedFile.from_upload(request.files['file'])
+        else:
+            data = loads(request.data)
+            if 'file' in data:
+                f = AttachedFile.from_upload(data['file'])
+            else:
+                abort(400)
 
-        f = AttachedFile.from_upload(request.files['file'])
         return render({'filename': url_for('api.Files:get', id=f.id)})
```

Comment:
> Surely this needs to be changed to AttachedFiles:get. 
Agreed.

> Also, why do we have an API to add an AttachedFile without attaching it to a node ?
This was added in e3d375741c9ca1e5859c4d647863d649866ca7a8 ("add markdown editor"), probably to be able to upload screenshots directly from the web UI